src/sentry/sentry_metrics/consumers/indexer/processing.py

```diff
@@ -1,18 +1,33 @@
 import logging
+from typing import Callable, Mapping
 
 from arroyo.types import Message
+from django.conf import settings
 
-from sentry.sentry_metrics import indexer
-from sentry.sentry_metrics.configuration import UseCaseKey
+from sentry.sentry_metrics.configuration import IndexerStorage, MetricsIngestConfiguration
 from sentry.sentry_metrics.consumers.indexer.batch import IndexerBatch
 from sentry.sentry_metrics.consumers.indexer.common import MessageBatch
+from sentry.sentry_metrics.indexer.base import StringIndexer
+from sentry.sentry_metrics.indexer.cloudspanner.cloudspanner import CloudSpannerIndexer
+from sentry.sentry_metrics.indexer.mock import MockIndexer
+from sentry.sentry_metrics.indexer.postgres.postgres_v2 import PostgresIndexer
 from sentry.utils import metrics
 
 logger = logging.getLogger(__name__)
 
+_MOCK = MockIndexer()
+_POSTGRES = PostgresIndexer()
+
+STORAGE_TO_INDEXER: Mapping[IndexerStorage, Callable[[], StringIndexer]] = {
+    # TODO(add cloudspanner options)
+    IndexerStorage.CLOUDSPANNER: lambda: CloudSpannerIndexer(**settings.CLOUDSPANNER_OPTIONS),
+    IndexerStorage.POSTGRES: lambda: _POSTGRES,
+    IndexerStorage.MOCK: lambda: _MOCK,
```

Comment:
to summarize the discussion in slack, i still think it would be cleaner if there was a single `IndexerStorage.PRODUCT` that uses the globally configured instance, instead of having an enum that has to enumerate all possible backends